src/gnc_attitude_estimation.cpp

```diff
@@ -45,33 +46,53 @@ AttitudeEstimate::AttitudeEstimate() {
  *  @returns Zero of success.
  *  Attempts to determine attitude given time, position, a sun vector, and
  *  magnetic field vector. The result, if succesful, is written into the
- *  quaternion q_body_eci. */
+ *  quaternion q_body_eci.
+ *  Prior to running triad, we ensure the following conditions are met:
+ *   - Magnitude of the expected magnetic field measurement is above the noise
+ *     floor of the magnetometer.
+ *   - Magnitude of the measured magnetic field is above the noise floor of the
+ *     magnetometer. */
 static int estimate_q_body_eci(double t, lin::Vector3d const &r_ecef,
     lin::Vector3f const &s_body, lin::Vector3f const &b_body, lin::Vector4f &q_body_eci) {
+
   // Setup the rest of the triad inputs (including expected sun and magnetic
   // field vectors in ECI)
-  lin::Vector3f s_eci, b_eci, r_ecef_f({
+  lin::Vector3f s_eci, b_eci, b_u_body, r_ecef_f({
```

Comment:
What's b_u?

Reply:
And just a unit vector (`b_body` is `const` so I had to make another variable for it). It's assigned like 5 lines down.